src/content/docs/docs/guides/vue.mdx

```diff
@@ -0,0 +1,755 @@
+const todos = computed(() => [...fetchedTodos.value].reverse());
+
+const todosCollection = bzr.collection<Todo>("todos");
+
+async function onSubmit() {
+ const unpersistedTodo = { text: todoText.value, completed: false };
+ const id = await todosCollection.insertOne(unpersistedTodo);
+ todos.value.push({ id, ...unpersistedTodo });
+ fetchedTodos.value.push({ id, ...unpersistedTodo });
+
+ todoText.value = "";
+}
+
+async function doLoginActions() {
+ isLoggedIn.value = true;
+ todosCollection.getAll().then((docs) => (todos.value = docs));
+ todosCollection.getAll().then((docs) => (fetchedTodos.value = docs));
+}
+
+...
+</script>
+```
+
+In these changes, you store fetched todos in a new reactive variable, `fetchedTodos`, and update `todos` to be a computed property that will updated when `fetchedTodos` changes. The spread operator (`...`) creates a new array, as the `reverse` method updates an array in place. Doing `fetchedTodos.value.reverse()` would cause an infinite re-render.
+
+You have now successfully displayed the todos fetched from the collection. In the next step, you will add realtime syncing to keep the todos updated across different devices.
+
+## Step 9 — Adding Realtime Syncing
+
+In this step, you will enable realtime updates for your todos so that changes are reflected instantly across different devices.
+
+The `todos` reactive variable is an array so you can use the `arrayMirrorSubscribeListener` helper. Import it:
+
+```vue title="App.vue" ins=", arrayMirrorSubscribeListener"
+import { BazaarApp, arrayMirrorSubscribeListener } from "@bzr/bazaar";
+```
+
+Then, update the `onSubmit` and `doLoginActions` methods as follows:
+
+```vue title="App.vue"
+<script setup lang="ts">
+...
+
+function onSubmit() {
+ const unpersistedTodo = { text: todoText.value, completed: false };
+ todosCollection.insertOne(unpersistedTodo);
+ todoText.value = "";
+}
+
+async function doLoginActions() {
+ isLoggedIn.value = true;
+ todosCollection.subscribeAll({}, arrayMirrorSubscribeListener(fetchedTodos.value));
+}
+
+...
+</script>
+```
+
+In this code, `todosCollection.subscribeAll` subscribes to all changes in the `todos` collection, and `arrayMirrorSubscribeListener` ensures that the local `todos` array stays in sync with the collection.
+
+You can write your own subscribe listeners. The syntax looks like this:
+
+```ts
+import type { AnyDoc, SubscribeListener } from "@bzr/bazaar";
+
+const mySubscribeListener: SubscribeListener<AnyDoc> = {
+  onInitial: (doc) => {
+    // Add doc
+ },
+  onAdd: (doc) => {
+    // Add doc
+ },
+  onChange: (oldDoc, newDoc) => {
+    // Update doc
+ },
+  onDelete: (doc) => {
+    // Delete doc
+ },
+};
+
+todosCollection.subscribeAll({}, mySubscribeListener)
+```
+
+The `arrayMirrorSubscribeListener` implements `onIntial` so that data from a collection is fetched and added to your local state. Essentially, it's syntactic sugar for `todosCollection.getAll().then((docs) => (todos.value = docs));`.
```

Comment:
Should we replace this with a note: to learn more about subscribe listeners go to docs link?

Reply:
My concern is that without showing a subscribe listener, we're presenting a whole lot of magic and giving little idea of how it works or how to modify it. But maybe it's actually overwhelming and counterproductive to see all of that. 

So ye, let's add a note, something like, "`arrayMirrorSubscribeListener` is a helper subscribe listener. Read the docs to learn how to create your own".

Reply:
Yeah, for a guide the magic is fine. It is supposed to be the first contact with bazaar.